### PR TITLE
feat: add artifact id and token endpoint

### PR DESCRIPTION
**Pull request by bwanglzu**
This PR allows user to get user token from Hubble, and get artifact id once fine-tuning is done. Usage:

```python
import finetuner

finetuner.login()
token = finetuner.get_token()

run = finetuner.get_run(run_name='something')
artifact_id = run.artifact_id
```

This allows user to directly pull artifact inside `FinetunerExecutor` using`token` and `artifact` as `artifact_id`.
This PR also did some minor refactoring on `download_artifact` and error handling.

A follow up PR will be created to update documentation including the usage of `FinetunerExecutor`.

---

- [ ] This PR references an open issue
- [x] I have added a line about this change to CHANGELOG

### CODE REVIEW

File: finetuner/__init__.py [hunk 1]
```diff
@@ -274,3 +274,12 @@ def delete_experiments() -> List[Experiment]:
     :return: List of deleted experiments.
     """
     return ft.delete_experiments()
+
+
+def get_token() -> str:
+    """Get user token.
+
+    This is a helper function to get token and pull artifact from cloud storage.
```
Comment (gmastrapas) [inline finetuner/__init__.py:282]: the docstring is confusing. This function does not download the artifact

File: finetuner/run.py [hunk 2]
```diff
@@ -56,18 +71,40 @@ def logs(self) -> str:
             experiment_name=self._experiment_name, run_name=self._name
         )
 
+    def _check_run_status(self):
+        status = self.status()[STATUS]
+        if status in [CREATED, STARTED]:
+            raise RunInProgressError(
+                'The run needs to be finished in order to save the model.'
```
Comment (gmastrapas) [inline finetuner/run.py:78]: save the artifact

**Review by gmastrapas — CHANGES_REQUESTED** [review]

File: finetuner/run.py [hunk 3]
```diff
@@ -56,18 +71,40 @@ def logs(self) -> str:
...
+        self._check_run_status()
         return download_artifact(
             client=self._client,
-            experiment_name=self._experiment_name,
+            artifact_id=self._run[ARTIFACT_ID],
             run_name=self._name,
             directory=directory,
         )
+
+    def artifact_id(self):
```
[the remark this answers is not in the document]
Comment (bwanglzu) [reply finetuner/run.py:99]: how about `staus()`?

[the remark this answers is not in the document]
Comment (gmastrapas) [reply finetuner/run.py:100]: well I think all should be properties 😛

[the remark this answers is not in the document]
Comment (bwanglzu) [reply finetuner/run.py:100]: i believe `status()` and `logs()` should be `get_`,  the rest could be property

**Review by gmastrapas — APPROVED** [review]

**Pull request merged**